### PR TITLE
Support test names with query strings

### LINKED ISSUE #362: Unable to see test results of tests with query parameters
1. Open https://wpt.fyi/results/dom
2. Click ```interfaces.html?exclude=Node```

Expected:
Showing test results of ```interfaces.html?exclude=Node```

Actual:
The new page URL is https://wpt.fyi/results/dom/interfaces.html, and the page doesn't contain any test results.

### PR DESCRIPTION
**Pull request by Hexcles**
## Description

Fixes #362 .

## Review Information

Navigate to `/dom/idlharness` and click around. Everything should work.

### CODE REVIEW

File: webapp/components/self-navigator.html [hunk 1]
```diff
@@ -11,17 +11,19 @@
             value: '/',
             observer: 'pathUpdated',
           },
+          encodedPath: {
+            type: String,
+            computed: 'computeEncodedPath(path)'
+          },
           onpathupdated: Function,
+          encodeTestPath: Function,
+          decodeTestPath: Function,
```
Comment (lukebjerring) [inline webapp/components/self-navigator.html]: I don't think these are needed; just always encode/decode the value as separated properties (`encodedPath` and `path`).

File: webapp/components/test-runs.html [hunk 2]
```diff
@@ -150,6 +150,25 @@
...
+        parts.push(encodeURIComponent(parts.pop()));
+        return '/' + parts.join('/');
+      }
+
+      decodeTestPath(path) {
+        console.assert(path.startsWith('/'));
+        let parts = path.split('/').slice(1);
+        parts.push(decodeURIComponent(parts.pop()));
+        return '/' + parts.join('/');
+      }
```
Comment (lukebjerring) [inline webapp/components/test-runs.html]: These functions can just be done on the base class; I can't imagine we'll ever decode things differently.

File: webapp/components/self-navigator.html (hunk 1 above)
Comment (Hexcles) [reply webapp/components/self-navigator.html]: Done.

File: webapp/components/test-runs.html (hunk 2 above)
Comment (Hexcles) [reply webapp/components/test-runs.html]: Done. Also, consequently, `test-file-results{,-time-series}.html` no longer call `encodeTestPath`, but take an encoded path as a property instead.

**Comment by Hexcles** [comment]
@lukebjerring can you take another look?

File: webapp/components/self-navigator.html [hunk 3]
```diff
@@ -11,17 +11,17 @@
             value: '/',
             observer: 'pathUpdated',
           },
+          encodedPath: {
+            type: String,
+            computed: 'computeEncodedPath(path)'
```
Comment (lukebjerring) [inline webapp/components/self-navigator.html]: Just use `encodeTestPath`, since that's the whole body of `computeEncodedPath`

**Pull request merged**